### PR TITLE
[gh-785] refactor to keep original Ethereum signatures, clean validators to keep rooch and ethereum validator

### LINKED ISSUE #785: [refactor] clean validators to keep rooch and ethereum validator
Part of #678

### PR DESCRIPTION
**Pull request by feliciss**
## Summary

1. Refactor Rooch signature to Ethereum signature. Keep each blockchain's original signature.
2. Clean validators, only keep Rooch native validator and Ethereum validator.

- Closes #785

### CODE REVIEW

File: crates/rooch-types/src/transaction/ethereum.rs [hunk 1]
```diff
@@ -74,33 +95,40 @@ impl EthereumTransaction {
...
-        // 99 length with 1 byte scheme following with 65 bytes recoverable signature and 33 bytes public key
-        let signature: Signature = <EcdsaRecoverableRoochSignature as ToFromBytes>::from_bytes(
-            &scheme_rsv_signature_pubkey,
-        )
-        .unwrap()
-        .into();
-
-        Ok(signature)
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct EthereumTransaction {
```
Comment (jolestar) [inline crates/rooch-types/src/transaction/ethereum.rs:117]: Why we need this struct?

Comment (feliciss) [reply crates/rooch-types/src/transaction/ethereum.rs:117]: I'm just copying the ones from the `RoochTransaction` in `rooch.rs` to facilitate passing `authenticator` as an arg to use in functions like `sign`, etc.

**Comment by feliciss** [comment]
I wanted to know why would the test fail with error OUT_OF_GAS:

> --- STDERR:              rooch-framework-tests tests::ethereum_validator_tests::test_validate ---
> thread 'tests::ethereum_validator_tests::test_validate' panicked at 'called Result::unwrap() on an Err value: status OUT_OF_GAS of type Execution', crates/rooch-framework-tests/src/tests/ethereum_validator_tests.rs:47:10
> note: run with RUST_BACKTRACE=1 environment variable to display a backtrace

File: crates/rooch-types/src/transaction/ethereum.rs (hunk 1 above)
Comment (jolestar) [reply crates/rooch-types/src/transaction/ethereum.rs:120]: The Ethereum original tx has the signature. We need to submit the original ethereum tx to the RPC.

File: crates/rooch-types/src/transaction/authenticator.rs [hunk 2]
```diff
@@ -187,59 +114,49 @@ where
...
-    pub scheme: u64,
+    pub coin_id: u64,
     pub payload: Vec<u8>,
 }
 
 impl Authenticator {
     /// Unique identifier for the signature scheme
-    pub fn scheme(&self) -> u64 {
-        self.scheme
+    pub fn coin_id(&self) -> u64 {
```
Comment (jolestar) [inline crates/rooch-types/src/transaction/authenticator.rs:143]: The coin_id here is a bit strange. Maybe confuse the user with the coin in the `coin.move`. But we can record an issue to discuss later.

@baichuan3 @wow-sven

File: crates/rooch-types/src/transaction/ethereum.rs (hunk 1 above)
Comment (feliciss) [reply crates/rooch-types/src/transaction/ethereum.rs]: Done.

File: crates/rooch-types/src/transaction/authenticator.rs (hunk 2 above)
Comment (feliciss) [reply crates/rooch-types/src/transaction/authenticator.rs:92]: `multichain_id` is fit cuz there's a `chain_id` in the genesis context.

Comment (feliciss) [reply crates/rooch-types/src/transaction/authenticator.rs:92]: I recorded an issue:

https://github.com/rooch-network/rooch/issues/791

I think we can just use custom chain id to build each blockchain ids by hand.

**Review by jolestar — APPROVED** [review]

**Pull request merged**